arch/src/x86_64/mod.rs

```diff
@@ -926,52 +926,16 @@ pub fn configure_system(
     )
 }
 
-fn configure_pvh(
-    guest_mem: &GuestMemoryMmap,
-    cmdline_addr: GuestAddress,
-    initramfs: &Option<InitramfsConfig>,
-    rsdp_addr: Option<GuestAddress>,
-    sgx_epc_region: Option<SgxEpcRegion>,
-) -> super::Result<()> {
-    const XEN_HVM_START_MAGIC_VALUE: u32 = 0x336ec578;
-
-    let mut start_info: StartInfoWrapper = StartInfoWrapper(hvm_start_info::default());
-
-    start_info.0.magic = XEN_HVM_START_MAGIC_VALUE;
-    start_info.0.version = 1; // pvh has version 1
-    start_info.0.nr_modules = 0;
-    start_info.0.cmdline_paddr = cmdline_addr.raw_value();
-    start_info.0.memmap_paddr = layout::MEMMAP_START.raw_value();
-
-    if let Some(rsdp_addr) = rsdp_addr {
-        start_info.0.rsdp_paddr = rsdp_addr.0;
-    }
-
-    if let Some(initramfs_config) = initramfs {
-        // The initramfs has been written to guest memory already, here we just need to
-        // create the module structure that describes it.
-        let ramdisk_mod: ModlistEntryWrapper = ModlistEntryWrapper(hvm_modlist_entry {
-            paddr: initramfs_config.address.raw_value(),
-            size: initramfs_config.size as u64,
-            ..Default::default()
-        });
-
-        start_info.0.nr_modules += 1;
-        start_info.0.modlist_paddr = layout::MODLIST_START.raw_value();
-
-        // Write the modlist struct to guest memory.
-        guest_mem
-            .write_obj(ramdisk_mod, layout::MODLIST_START)
-            .map_err(super::Error::ModlistSetup)?;
-    }
-
-    // Vector to hold the memory maps which needs to be written to guest memory
-    // at MEMMAP_START after all of the mappings are recorded.
-    let mut memmap: Vec<hvm_memmap_table_entry> = Vec::new();
-
-    // Create the memory map entries.
-    add_memmap_entry(&mut memmap, 0, layout::EBDA_START.raw_value(), E820_RAM);
+type RamRange = (u64, u64);
 
+/// Returns usable physical memory ranges for the guest
+/// These should be used to create e820_RAM memory maps
+///
+/// There are up to two usable physical memory ranges,
+/// devidided by the gap at the end of 32bit address space.
```

Comment:
nit: :s/devidided/divided

Reply:
Good catch. I am a bit surprised that our GH action does not catch this typo.